input/script_utils.go

```diff
@@ -1792,23 +2011,69 @@ func CommitScriptToSelf(csvTimeout uint32, selfKey, revokeKey *btcec.PublicKey)
 // key, or a NUMs point for the remote output) along with the tapscript leaf
 // that can spend the output after a delay.
 type CommitScriptTree struct {
-	// TaprootKey is the key that will be used to generate the taproot
-	// output.
-	TaprootKey *btcec.PublicKey
+	ScriptTree
 
 	// SettleLeaf is the leaf used to settle the output after the delay.
 	SettleLeaf txscript.TapLeaf
 
 	// RevocationLeaf is the leaf used to spend the output with the
 	// revocation key signature.
 	RevocationLeaf txscript.TapLeaf
+}
 
-	// TapscriptTree is the full tapscript tree that also includes the
-	// control block needed to spend each of the leaves.
-	TapscriptTree *txscript.IndexedTapScriptTree
-
-	// TapscriptTreeRoot is the root hash of the tapscript tree.
-	TapscriptRoot []byte
+// A compile time check to ensure CommitScriptTree implements the
+// TapscriptDescriptor interface.
+var _ TapscriptDescriptor = (*CommitScriptTree)(nil)
+
+// WitnessScript returns the witness script that we'll use when signing for the
+// remote party, and also verifying signatures on our transactions. As an
+// example, when we create an outgoing HTLC for the remote party, we want to
+// sign their success path.
+func (c *CommitScriptTree) WitnessScriptToSign() []byte {
+	// TODO(roasbeef): abstraction leak here? always dependent
```

Comment:
what did you mean?

Reply:
In that this distinction matters more for the HLTC scripts, but I still found it to be useful elsewhere to simplify code. When signing for the remote party, it's effectively always a keyspend for us.